tests/test_writer.py

```diff
@@ -1202,25 +1202,42 @@ def test_attachments():
     b.seek(0)
     reader = PdfReader(b)
     b = None
+    assert reader.attachments == {}
     assert reader._list_attachments() == []
     assert reader._get_attachments() == {}
-    writer.add_attachment("foobar.txt", b"foobarcontent")
-    writer.add_attachment("foobar2.txt", b"foobarcontent2")
-    writer.add_attachment("foobar2.txt", b"2nd_foobarcontent")
+    to_add = [
+        ("foobar.txt", b"foobarcontent"),
+        ("foobar2.txt", b"foobarcontent2"),
+        ("foobar2.txt", b"2nd_foobarcontent"),
+    ]
+    for name, content in to_add:
+        writer.add_attachment(name, content)
 
     b = BytesIO()
     writer.write(b)
     b.seek(0)
     reader = PdfReader(b)
     b = None
-    assert reader._list_attachments() == ["foobar.txt", "foobar2.txt", "foobar2.txt"]
+    assert sorted(reader.attachments.keys()) == sorted({name for name, _ in to_add})
```

Comment:
is the sorted necessary : I prefer to check that the order is kept

Reply:
`reader.attachments` is a dictionary which doesn't have an order.

Do the order of attached files have a meaning?

Reply:
Could be : you can have version 1, version 2, ... with the same name

Reply:
If it's the same name then it would only be one key in `reader.attachments.keys()`

Reply:
sorry Just seen my misunderstanding. All good!